### DISCUSSION TITLE
Very weird routing issue

### DISCUSSION THREAD

**Discussion by 9bits** [vercel/next.js#17436] [Help]
I have encountered a very peculiar routing issue.

My app currently has the following file structure

```
/
    /featured
        [id].tsx
    featured.tsx
```

the [id.tsx] currently retrieves the query param and displays it.
```
export type IndexProps = {
  query: string;
};

function Index({ query }: IndexProps) {
  return <div>{query}</div>;
}

export const getServerSideProps: GetServerSideProps = async (context) => {
  const { req, query, res } = context;

  return {
    props: {
      query: query.id,
    },
  };
};
export default Index;
```

However, I have encountered an issue where for SOME query params, the page keeps redirecting me back to the featured.tsx page instead of the [id].tsx

I was doing some prior development and I feel like there is some uncleared cache that is causing this.

Is there a way to clean up the cache for which nextjs uses for routing?

Otherwise, I appreciate other suggestions. 

Thanks

OS: Windows 10
Browser: Microsoft Edge

I just tested in Chrome and it seems to work as expected. I think I need to clear the cache in Edge

**Comment by jamesmosier** [comment]
Hi @9bits. There shouldn't be any issue with caching when navigating to your different routes. Could you provide an example of what routes you were trying to visit and with the url parameters?
